### ISSUE TITLE
Lifeline extends a little past deactivation (teoz engine)

### ISSUE THREAD

**Issue by jhoehle** [plantuml/plantuml#1094]
This is the shortest test case about a broken lifeline length that I came across:
![](https://www.plantuml.com/plantuml/png/bP5DRi8m44RN-1GVBDX4KVUHGY6gL5NY2DaOCXWBPoQr9_tfwPjK8YAYNVG2xytxPflKHktRO9PTjAwrK98lQEp9b2Kgs-05PMbso3Uea6MBHO7EimDWM9GGlJjhWRq_KV1Ce0ybRbBDw6nIj9IITMGgvFdc4GM2EBD6DV86QFx34tQJxLQtkBNjZkhVh1AgcPXUUnl29mvHBiGrltizIwy3-Y7BmA9MlV0V_id-H7R2ZwSyYpI_Ti_SXrv70pgA9pfgKVDgmA-W8ZW6ILGpL12ksy3STwynEssSWPB-SuIzo2LTprZEOsTfz-LRucRy_pS0)
```puml
@startuml
!pragma teoz true
== Bad I ==
A -> B ++		: ping   
note right: Lifeline extends\n past message
B -->? : log
& B -> A --		: pong
== Bad II ==
A -> B 			: ping   
Activate B
note right: equally broken\nwithout --/++ notation
B -->? : log
& B -> A 		: pong
Deactivate B
== Good ==
A -> B ++		: ping
note right: but not perfect:\n"log" too close\n to lifeline  
B -> A --		: pong
& B -->? : log
== Best ==
A -> B 			: ping   
Activate B
note right: Looks perfect!
B -->? : log
Deactivate B
& B -> A 		: pong
@enduml
```
Changing the order magically works-around the issue in this particular example!

Lifelines with wrong offsets or lengths are observable in many pictures across the net, but few users bother to report a bug.
 - Bug plantuml/plantuml#656
 - https://forum.plantuml.net/13409/teoz-arrow-alignment (lifeline starts too late)
 - https://forum.plantuml.net/12899/sequence-small-vertical-offset-before-next-messsage-arrow
 - Whereas bug plantuml/backlog#9 is different, a matter of visual cue IMHO, given the particular shape of the self-referential arrow.

**Comment by jhoehle** [comment]
Edited description to add a fourth test case.

**Comment by chipbite** [comment]
Great bug report! Here is another bug that I suspect may be related to this, i e activation and teoz - this: 
* #739, also reported here: https://forum.plantuml.net/14873/teoz-activation-breaks-parallell-lines

**Comment by jimnelson372** [comment]
Hi @jhoehle, 

The lifeline issue you demonstrate in your example has been resolved in the latest release, 1.2024.5.   Here is what your exact script now generates:

![testteozaltparallel_008](https://github.com/plantuml/plantuml/assets/66284321/f323d8e8-4a87-4e8f-b974-9daf1c79dc03)

I had used your example while working on this and related issues, but overlooked this issue # when I submitted the fix in pull request #1790.  

Regards,
Jim Nelson

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed.
- Thanks @jhoehle  for the report.
- Thanks PlantUML team and @jimnelson372 for your work.

_So do no hesitate to reopen if there is issue on that!_

Regards.
